tests/unit/einstein/test_ceph.py

```diff
@@ -0,0 +1,141 @@
+
+    @classmethod
+    def tearDownClass(self):
+        self.fs.remove(CEPH_IMG)
+
+
+class TestSnapshot(unittest.TestCase):
+    """Test snapshot operations"""
+
+    @classmethod
+    def setUpClass(self):
+        self.fs = ceph.RBD(_cfg.fs, _cfg.iscsi.password)
+        self.fs.create_image(CEPH_IMG, CEPH_IMG_SIZE, False, 1)
+        self.fs.snap_image(CEPH_IMG, CEPH_SNAP_IMG)
+
+    def test_01_snap_image(self):
+        """Test create snapshot"""
+
+        snapList = self.fs.list_snapshots(CEPH_IMG)
+        self.assertIn(CEPH_SNAP_IMG, snapList)
+
+    def test_02_image_exist_exception(self):
+        self.assertRaises(file_system_exceptions.ImageExistsException,
+                          self.fs.snap_image, CEPH_IMG, CEPH_SNAP_IMG)
+
+    def test_03_snap_protect(self):
+        """Test protecting a snapshot"""
+
+        protectFlag = self.fs.snap_protect(CEPH_IMG, CEPH_SNAP_IMG)
+        self.assertTrue(protectFlag, "Failed to protect a snapshot!")
+
+    def test_04_clone(self):
+        """Test clone image"""
+
+        cloneFlag = self.fs.clone(CEPH_IMG, CEPH_SNAP_IMG,
+                                  CEPH_CHILD_IMG)
+        self.assertTrue(cloneFlag, "Failed to clone image")
+        children = self.fs.list_children(CEPH_IMG, CEPH_SNAP_IMG)
+        children_imgs = [child[1] for child in children if child[0] == POOL]
+        self.assertIn(CEPH_CHILD_IMG, children_imgs,
+                      "Error, cannot find child image")
+        parent_info = self.fs.get_parent_info(CEPH_CHILD_IMG)
+        self.assertEqual(CEPH_IMG, parent_info[1])
+        self.assertEqual(CEPH_SNAP_IMG, parent_info[2])
+        self.fs.flatten(CEPH_CHILD_IMG)
+
+    def test_05_snap_unprotect(self):
+        """Test unprotecting a snapshot"""
+
+        unprotectFlag = self.fs.snap_unprotect(CEPH_IMG, CEPH_SNAP_IMG)
+        self.assertTrue(unprotectFlag, "Failed to unprotect a snapshot")
+
+    def test_06_remove_snapshot(self):
+        """Test Remove Snapshot"""
+
+        self.assertRaises(file_system_exceptions.ImageHasSnapshotException,
+                          self.fs.remove, CEPH_IMG)
+        self.fs.remove_snapshot(CEPH_IMG, CEPH_SNAP_IMG)
+        snapshots = self.fs.list_snapshots(CEPH_IMG)
+        self.assertNotIn(CEPH_SNAP_IMG, snapshots,
+                         "Failed to remove a snapshot")
+        # remove parent image, note here remove order does not matter
+        self.fs.remove(CEPH_IMG)
+        imgs = self.fs.list_images()
+        self.assertNotIn(CEPH_IMG, imgs, "Failed to remove test parent image!")
+        # remove child image
+        self.fs.remove(CEPH_CHILD_IMG)
+        imgs = self.fs.list_images()
+        self.assertNotIn(CEPH_CHILD_IMG, imgs, "Failed to remove test child \
+                                                image clone from parent image \
+                                                %s!" % CEPH_IMG)
+
+    @classmethod
+    def tearDownClass(self):
```

Comment:
Again seeing the numbering and everything you thought that the tests will run in top down manner. Each test case should be independent and assume can run at any order. Same comment as before so change to setUp/tearDown.